### PR TITLE
feat(emacs)!: update script to latest spec

### LINKED ISSUE #1303: Bug: Emacs package missing dependencies
The emacs package appears to be missing some dependencies needed to make it compile (at least on Ubuntu 20.04). The current AUR package has a number of other dependencies: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=emacs28-git

The following are what I needed to get this compiled (or at least to start compiling - will update if anything more is needed)
- libgtk-3-dev
- libxpm-dev
- libgif-dev

### PR DESCRIPTION
**Pull request by wizard-28**
Updates the pacscript to the latest PKGBUILD spec
https://aur.archlinux.org/cgit/aur.git/commit/?h=emacs-git&id=b9eb3b405717f8df37b38217dbb7d361b8ffca08

Closes #1303

### CODE REVIEW

**Comment by wizard-28** [comment]
@JonBoyleCoding Please test if this works.

**Comment by wizard-28** [comment]
Also @Henryws this is a breaking change for the pacscript as a lot of deprecated features have been removed from it

**Comment by wizard-28** [comment]
@Henryws The CI doesn't work, jitterbit fails.

**Review by Henryws — CHANGES_REQUESTED** [review]
image

**Comment by JonBoyleCoding** [comment]
Sorry I haven't had the time to test this out. I won't have time until end of next week to do anything on the side - if you still need some help with this by then I can have a look then.

File: packages/emacs/emacs.pacscript [hunk 1]
```diff
@@ -91,57 +113,69 @@ if [[ $LTO == "YES" ]]; then
...
 
-if [[ $JIT == "YES" ]]; then
-  if [[ $CLI == "YES" ]]; then
-    depends_nox+=" libgccjit0"
+if [[ ${JIT} == "YES" ]]; then
+  if [[ ${CLI} == "YES" ]]; then
+    depends_nox+=" libgccjit0 libgccjit-11-dev"
   else
-    depends+=" libgccjit0"
+    depends+=" libgccjit0 libgccjit-11-dev"
```
Comment (JonBoyleCoding) [inline packages/emacs/emacs.pacscript:120]: For Ubuntu 20.04 - needs to be libgccjit-10-dev as 11+ isn't in the repository

File: packages/emacs/emacs.pacscript [hunk 2]
```diff
@@ -91,57 +113,69 @@ if [[ $LTO == "YES" ]]; then
...
+  build_depends+=" x11proto-dev libxi-dev"
 fi
 
-if [[ $MAGICK == "YES" ]]; then
-  depends+=" imagemagick libjpeg-turbo8 giflib-tools"
-elif [[ $NOX != "YES" ]] && [[ $CLI != "YES" ]]; then
-  depends+=" libjpeg-turbo8 giflib-tools"
-elif [[ $CLI == "YES" ]]; then
+if [[ ${CLI} != "YES" ]]; then
+  depends+=" libjpeg-turbo8 libpng16-16 libgif-dev libwebp7 libtiff5 libxpm-dev"
```
Comment (JonBoyleCoding) [inline packages/emacs/emacs.pacscript:138]: libwebp7 needs to be libwebp6 for Ubuntu 20.04

**Review by JonBoyleCoding — COMMENTED** [review]
I am testing for Ubuntu 20.04 as that is what I am limited to by my organisation for the time being. If you wish to support that still, then the following packages will need to be used instead of the ones currently in the package.

If changed, the pacscript seems to install fine on 20.04. I would propose the change myself for multiple distros, but I don't know if you wish to go down that path or what the best practice would be.

**Comment by Henryws** [comment]
@wizard-28 can you look at the reviews?

**Comment by wizard-28** [comment]
For Ubuntu:

```
emacs : Depends: libgtk-3-dev but it is not going to be installed
         Depends: libwebp6 but it is not installable
```

For Debian:

```
 emacs : Depends: libjpeg-turbo8 but it is not installable
         Depends: libwebp6 but it is not installable
```

**Comment by wizard-28** [comment]
`libwebp7` seems to be present in `ubuntu:latest`, but not `libwebp6`

`libgtk-3-dev` is present in `ubuntu:latest`, but not in `ubuntu:devel` and `ubuntu:rolling`

**Comment by wizard-28** [comment]
@Henryws what to do?

**Comment by Henryws** [comment]
Maybe use incompatible

**Comment by wizard-28** [comment]
> Maybe use incompatible

Should I add OS-specific dependencies?

**Comment by Henryws** [comment]
Sure

**Review by Henryws — APPROVED** [review]

**Pull request merged**